lib/parallel_cucumber/worker.rb

```diff
@@ -291,18 +291,28 @@ def setup(env)
     def parse_results(f, tests)
       unless File.file?(f)
         @logger.error("Results file does not exist: #{f}")
-        return tests.map { |t| [t, ::ParallelCucumber::Status::UNKNOWN] }.to_h
+        return unknown_result(tests)
       end
       json_report = File.read(f)
       if json_report.empty?
         @logger.error("Results file is empty: #{f}")
-        return tests.map { |t| [t, ::ParallelCucumber::Status::UNKNOWN] }.to_h
+        return unknown_result(tests)
       end
       Helper::Cucumber.parse_json_report(json_report)
     rescue => e
       trace = e.backtrace.join("\n\t").sub("\n\t", ": #{$ERROR_INFO}#{e.class ? " (#{e.class})" : ''}\n\t")
       @logger.error("Threw: JSON parse of results caused #{trace}")
-      tests.map { |t| [t, ::ParallelCucumber::Status::UNKNOWN] }.to_h
+      unknown_result(tests)
+    end
+
+    private
+
+    def unknown_result(tests)
+      res = {}
+      tests.each do |test|
+        res[test.to_sym] = { status: ::ParallelCucumber::Status::UNKNOWN }
+      end
+      res
```

Comment:
```suggestion
      tests.map do |test, status|
        [test.to_sym, { status: ::ParallelCucumber::Status::UNKNOWN }].
      end.to_h
```
or, if you like
```suggestion
      res = tests.map do |test, status|
        [test.to_sym, { status: ::ParallelCucumber::Status::UNKNOWN }].
      end
      res.to_h
```

And also I would move `unknown_result` to `Helper::Cucumber` to make it clear why do we have this format

Reply:
```status``` variable in the block is not used, so ```_status``` :-)

Reply:
> `status` variable in the block is not used, so `_status` :-)

It's not needed actually. Fixed the comment

Reply:
Moved

Reply:
but. yeah. the ```map``` variant is definitely a way to go